src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -447,9 +502,16 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             Arguments.ManifestResources,
                             emitOptions,
                             debugEntryPoint: null,
+                            testData: null,
                             getHostDiagnostics: getAnalyzerDiagnostics,
                             cancellationToken: cancellationToken);
 
+                        if (!emitResult.Success && (newCompilation != null) && (newCompilation != compilation))
+                        {
+                            compilation = newCompilation;
+                            goto retry;
```

Comment:
where is this label?  Is there no way to use simpler codeflwo constructs vs using a goto?  also, can you comment why this chek is even neeeded?